desafio-07/kortkamp/c/tac.c

```diff
@@ -0,0 +1,93 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdio.h>
+#define LINE_BUFFER_SIZE 100000 
+#define BUFFER_SIZE 1000000
+
+/* TODO LIST	
+ * -need to deal with lines bigger then LINE_BUFFER_SIZE(yes, im noob)
+ * -print the line using the data stored in buffer, not loading linebuffer
+ * -use malloc instead of arrays 
+ * 
+ *
+ */
+
+FILE *fptr;
+char buffer[BUFFER_SIZE] = {0};
+char linebuffer[LINE_BUFFER_SIZE+11111] = {0};
+
+
+void clear_buffer(){
+	for(int i = 0; i <= BUFFER_SIZE; i++)
+	       buffer[i] = 0;	
+
+}
+//print line delimited by file indexes : start , end
+//Precisa preservar o SEEK_CUR no arquivo 
+int print_line(long start, long end){
+	long cur_pos = ftell(fptr);
+	
+	fseek(fptr,start,SEEK_SET);
+	if((end - start) <= LINE_BUFFER_SIZE){
+
+		fread(&linebuffer, sizeof(char),end - start, fptr);
+		linebuffer[end - start] = 0;
+		printf("%s\n",linebuffer);
+	}
+
+	// precisa retornar pra onde o ponteiro do arquivo estava antes
+	fseek(fptr,cur_pos,SEEK_SET);
+	return(0);
+}
+
+int main(int argc , char* argv[]){
+	long pos = LINE_BUFFER_SIZE-1 ;
+	long read_ret;
+	long seek_ret;
+	int last_bytes = BUFFER_SIZE; //last bytes to read when fseek reaches SEEK_SET
+	long fsize = 0;
+	long last_cr = 0;
+	if(argc <= 1){
+		printf("Usage: %s path_to_file\n",argv[0]);
+		return(0);
+	}
+	fptr = fopen(argv[1],"rb+");
+	if(fptr == NULL){
+		printf("Error opening file");
+		return(1);
+	}
+	//dispensa o último char do arquivo porque costuma ser convertido em \n
```

Comment:
E se não for? O que acontece?

Reply:
se não for , ele continua normalmente o programa.